src/modules/username.rs

```diff
@@ -0,0 +1,43 @@
+use ansi_term::Color;
+use std::env;
+use std::process::Command;
+
+use super::{Context, Module};
+
+/// Creates a segment with the current user's username
+///
+/// Will display the Node.js version if any of the following criteria are met:
+///     - The current user isn't the same as the one that is logged in ($LOGNAME != $USER)
+///     - The current user is root (UID = 0)
+///     - The user is currently connected as an SSH session ($SSH_CONNECTION)
+pub fn segment(_context: &Context) -> Option<Module> {
+    let user = env::var("USER").ok();
+    let logname = env::var("LOGNAME").ok();
+    let ssh_connection = env::var("SSH_CONNECTION").ok();
+    let uid = get_uid();
+
+    if user != logname || uid == Some(0) || ssh_connection.is_some() {
```

Comment:
Because `get_uid()` spawns a new process, let's try and make that the last check in this series of checks.

```suggestion
    if user != logname || ssh_connection.is_some() || is_root() {
```

In this example, `is_root()` would perform the full uid check so that we don't need to `get_uid()` if either of the other two are successful. You can maybe also change the module color within that function.